internal/gatewayapi/listener.go

```diff
@@ -119,6 +119,13 @@ func (t *Translator) ProcessListeners(gateways []*GatewayContext, xdsIR XdsIRMap
 					irListener.Hostnames = append(irListener.Hostnames, "*")
 				}
 				gwXdsIR.HTTP = append(gwXdsIR.HTTP, irListener)
+			case v1beta1.TCPProtocolType, v1beta1.TLSProtocolType:
```

Comment:
dont we need to set `irListener.Hostnames` for `TLS` ?
https://gateway-api.sigs.k8s.io/references/spec/#gateway.networking.k8s.io%2fv1beta1.Listener